### ISSUE TITLE
[java] ArrayIsStoredDirectly should report the assignment rather than formal parameter

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#3929]
**Is your feature request related to a problem? Please describe.**
The rule [ArrayIsStoredDirectly](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#arrayisstoreddirectly) reports the formal parameter of the method / constructor which is an array and which is stored directly.

It would be more useful, if the rule reported where the array is stored.

Example:

```java
public class ExampleAISD { 

    public String[] strs;

    public void func(String[] strs, boolean flag) { // line 5
        if (flag) {
            this.strs = strs; // line 7
        }
    }
}
```

Instead of line 5, line 7 should be reported.

**Describe the solution you'd like**

Report the assignment node instead.

**Describe alternatives you've considered**
None.

**Additional context**
See #3879 for original request
